go/libkb/merkle_client.go

```diff
@@ -720,58 +1055,66 @@ func (mc *MerkleClient) LookupUser(ctx context.Context, q HTTPArgs, sigHints *Si
 	mc.G().Log.CDebugf(ctx, "+ MerkleClient.LookupUser(%v)", q)
 
 	var path *VerificationPath
+	var ss SkipSequence
+	var apiRes *APIRes
 
 	if err = mc.init(ctx); err != nil {
-		return
+		return nil, err
 	}
 
 	// Grab the cached seqno before the call to get the next one is made.
 	// Note, we can have multiple concurrenct calls to LookupUser that can return in any order.
 	// Checking against the cache after the call completes can cause false-positive rollback
 	// warnings if the first call is super slow, and the second call is super fast, and there
 	// was a change on the server side. See CORE-4064.
-	seqnoBeforeCall := mc.LastSeqno()
+	rootBeforeCall := mc.LastRoot()
 
-	mc.G().Log.CDebugf(ctx, "| LookupPath")
-	if path, err = mc.LookupPath(ctx, q, sigHints); err != nil {
-		return
+	if path, ss, apiRes, err = mc.lookupPathAndSkipSequence(ctx, q, sigHints, rootBeforeCall); err != nil {
+		return nil, err
+	}
+
+	// It's important to check the merkle skip sequence before verifying the root.
+	mc.G().Log.CDebugf(ctx, "| VerifySkipSequence")
+	if err = mc.verifySkipSequence(ctx, ss, path.root, rootBeforeCall); err != nil {
+		mc.G().Log.CDebugf(ctx, "| Full APIRes was: %s", apiRes.Body.MarshalToDebug())
+		return nil, err
 	}
```

Comment:
Would it be alright to put `verifySkipSequence` _inside_ of `lookupPathAndSkipSequence`? It's important that you never do the first without the second, but I imagine someone's going to want to come back here and add new entrypoints that don't deal directly with loading a user.

Also maybe same question about the signature checking step immediately below?